_posts/2024-08-16-AI-chatting.md

```diff
@@ -0,0 +1,102 @@
+* **Processing by the LLM:** The chosen AI service provider or the separate application will process the embeddings via a LLM (or a similar technology) and send a response back to JabRef, which is then displayed in the UI.
+
+## Which AI provider should I use?
+
+> [!Note]
+> To use any of the providers, it is required to configure an API key in JabRef's preferences.
+We have many integrations with AI providers, so you can choose the one you like the most 😊 
+
+There are various considerations, such as pricing, capabilities, speed, privacy policy and more (that may change at any time) to take into account that make it difficult to give a clear recommendation that will stand the test of time. I strongly recommend to look up the models pricing before you choose. Smaller models will be cheaper and faster than larger models, whereas larger models tend to outperform in terms of capabilities. In any case, here is the list of providers:
+
+[OpenAI](https://platform.openai.com/docs/models).
+[MistralAI](https://docs.mistral.ai/getting-started/models/)
+[Huggingface](https://huggingface.co/models?pipeline_tag=text-generation&sort=trending)
+
+
+While OpenAI deserved a spot because of its popularity, Mistral's models are high up there in the benchmarks too and they led the way in terms of releasing [open weight](https://github.com/Open-Weights/Definition) models with Apache 2.0 licenses, which should warm the heart of anybody favouring Open Source. Hugging Face is currently a little bit special, as they give you access to thousands of open weight models [for free](https://huggingface.co/docs/api-inference/index), including ones with commercially friendly licenses (Apache 2.0, MIT), however, it may take a long time for Hugging Face to allocate free computing resources, resulting in longer response times, unless you choose (and pay) for a [GPU space](https://huggingface.co/docs/hub/spaces-gpus) that is much faster at inference.
+
+Finally, iF you are asking yourself, where can I find an API key and what is it even? Please head to the [AI user documentation](https://docs.jabref.org/ai/ai-providers-and-api-keys) to learn about how to receive a key and where to enter it in the preferences.
+
+## AI preferences
+
+Yes, by the way, there are some new options in the JabRef preferences, such as
+
+- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option if you want to use the new AI features.
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face.
+- "Chat model": choose the model you like
+- "API token": here you write your API token.
+- "Expert settings": the defaults provide good results. In case you would like to modify them, our [user documentation](https://docs.jabref.org/ai) provides an explanation of it.
+
+![AI preferences](../img/AiPreferences.png)
+
+## BONUS
+
+Have you ever considered running a local LLM on your computer while ensuring complete privacy?
```

Comment:
If you are considering running a local LLM on your computer to ensure complete privacy, check out my tutorial ...